src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricSettingsCmdletBase.cs

```diff
@@ -0,0 +1,142 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ServiceFabric.Models;
+using Microsoft.Azure.Management.ServiceFabric.Models;
+
+namespace Microsoft.Azure.Commands.ServiceFabric.Commands
+{
+    public class ServiceFabricSettingsCmdletBase : ServiceFabricClusterCmdlet
+    {
+        private List<PSSettingsSectionDescription> changedSettingsSectionDescriptions =
+            new List<PSSettingsSectionDescription>();
+        protected const string OneCertSet = "OneSetting";
+        protected const string GroupCertSet = "BatchSettings";
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = OneCertSet,
+                   HelpMessage = "Section")]
+        [ValidateNotNullOrEmpty()]
+        public string Section { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = OneCertSet,
+                   HelpMessage = "Parameter")]
+        [ValidateNotNullOrEmpty()]
+        public string Parameter { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = OneCertSet,
+                   HelpMessage = "Value")]
+        [ValidateNotNullOrEmpty()]
+        public virtual string Value { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = GroupCertSet,
+                   HelpMessage = "Client authentication type")]
+        [ValidateNotNullOrEmpty()]
+        public PSSettingsSectionDescription[] SettingsSectionDescriptions { get; set; }
```

Comment:
it is complex object , not sure we have a better way to represent it in Power Shell 

 public class SettingsSectionDescription
    {
        public SettingsSectionDescription();
        public SettingsSectionDescription(string name, IList<SettingsParameterDescription> parameters);

        [JsonProperty(PropertyName = "name")]
        public string Name { get; set; }
        [JsonProperty(PropertyName = "parameters")]
        public IList<SettingsParameterDescription> Parameters { get; set; }

        public virtual void Validate();
    }

 public class SettingsParameterDescription
    {
        public SettingsParameterDescription();
        public SettingsParameterDescription(string name, string value);

        [JsonProperty(PropertyName = "name")]
        public string Name { get; set; }
        [JsonProperty(PropertyName = "value")]
        public string Value { get; set; }

        public virtual void Validate();
    }